cirq/pasqal/pasqal_device.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2020 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import FrozenSet, Iterable, cast
+from numpy import sqrt
+
+import cirq
+
+from cirq.pasqal import ThreeDGridQubit
+
+
+@cirq.value.value_equality
+class PasqalDevice(cirq.devices.Device):
+    """A Pasqal Device with qubits placed on a 3D grid."""
+
+    def __init__(self, control_radius: float,
+                 qubits: Iterable[ThreeDGridQubit]) -> None:
+        """ Initializes the description of the device.
+
+        Args:
+            control_radius: the maximum distance between qubits for a controlled
+                gate. Distance is measured in units of the indices passed into
+                the qubit constructor.
+            qubits: Qubits on the device, identified by their x, y, z position.
+                Must be of type ThreeDGridQubit.
+
+        Raises:
+            ValueError: if the wrong qubit type is provided or if invalid
+                parameter is provided for control_radius. """
+        us = cirq.value.Duration(micros=1)
```

Comment:
If there is no urgent reason to use `micros`, let's use `millis` so we don't have to multiply by 5000